### ISSUE TITLE
Severe interface slow

### ISSUE THREAD

**Issue by DCNick3** [clementine-player/Clementine#6406]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Archlinux
- Clementine version: 1.3.1-704-g9af827b6a

### Expected behaviour / actual behaviour

Clementine is very slow at everything (startup, starting to play a song, create playlist, etc), when the data folder (supposedly, I have all volumes in btrfs) is stored in btrfs. It affects only interface, song playback is OK, but every interaction with interface make clementine freeze in I/O for ~1 secound. This is probably due to very slow `fsync` in btrfs - dpkg has the same problem, as it does a lot of those. Checking with `eatmydata`, which effectively disables `fsync`'s and makes clementine fast again.

### Steps to reproduce the problem (only for bugs)

1. Install linux to btrfs
2. Run clementine there

**Comment by aerusso** [comment]
Please try with qt5 master. #6398 addressed a lot of these issues, and was very recently accepted.

**Comment by DCNick3** [comment]
At first glance issue looks like resolved. Will do further testing.

**Comment by DCNick3** [comment]
Looks like this fix makes clementine responsiveness better. The freeze for `fsync` is still there, but it is delayed by one second. The improvement is that it does less `fsync`'s if I actively use interface.

**Comment by aerusso** [comment]
Yeah, there's got to be at least one fsync, short of some significant architectural change to the QSetting object that we use (that's a QT issue).

The framework I introduced in that patch lets us defer the save by that one second, allowing a bunch of changes to be made in the same transaction (resizing windows, with hundreds of updates was *miserable* before it, as I'm sure you're aware!). The fsync is done synchronously (hence the hang), after this delay. One solution would be to perform the sync in another thread, but that introduces a bunch of other problems (e.g., QSettings isn't thread safe, so you'd have to rotate the QSettings object. I haven't thought this through, but it might be relatively simple).

The lowest-hanging fruit at this point would be to indentify the other settings changes that you're noticing (e.g., add a list to the favorites), and switch them over to the new deferred framework. There are a TON of them (grep the code base for QSettings, most creations of QSettings objects should be replaced by using the deferred framework).  To do so, you have to get the global QSettings object, trigger the DirtySettings event and add a listener to the SaveSettings event from Application.  Having the app object reference means you have to trace the child object creation, and pass the app object from parent to child there (possibly multiple times). I didn't try to get all of them (or even most of them), I don't have the time.

@DCNick3  If you've got inclination, you could help out with a patch! I can help review the patch, and answer questions you might have. (But I've got too much stuff on my plate to write code for Clementine right now).

**Comment by jonaski** [comment]
I think we could increase the timer a bit
